### ISSUE TITLE
Problem with unaligned writes!

### ISSUE THREAD

**Issue by ju6ge** [Terah-/rust-bytebuffer#15]
Hi I am trying to use bytebuffer to implement Spi communication with a target device.

While doing so I have hit the following issue.

```rust
use bytebuffer::ByteBuffer;

fn main() {
    let mut bytes = ByteBuffer::new();
    bytes.write_u8(0x2);
    bytes.write_u16(0);
    bytes.write_bits(0x4, 3);
    bytes.write_bits(7, 5);
    println!("{}", bytes.to_hex_dump());
}
```
The result of running this code is `0x02 0x00 0x00 0x83 0xc0`.

But that is not the expected result! The result should only be 4 bytes long, not 5! The expected value would be `0x02 0x00 0x00 0x87`!

So it seems to me that there is an issue with the logic of appending single bit values …

**Comment by nobodie** [comment]
Hello,
there was indeed an issue with the logic. I have proposed a pull request to fix it [here](https://github.com/terahlunah/bytebuffer/pull/16)

Basically, the write_bits method was writing one more bit at the end, resulting in a wrong bit offset for the next write_bits call.

**Comment by terahlunah** [comment]
Thank you @ju6ge for the report and thank you @nobodie for the PR!

I published a new version with the fix : `2.1.1`

**Comment by ju6ge** [comment]
Awesome, props for the fast response and fix :partying_face:
